pyscriptjs/src/components/pyscript.ts

```diff
@@ -68,162 +68,144 @@ export function make_PyScript(interpreter: InterpreterClient, app: PyScriptApp)
+    "copy", "cut", "paste", "abort", "canplay", "canplaythrough", "cuechange",
+    "durationchange", "emptied", "ended", "loadeddata", "loadedmetadata",
+    "loadstart", "pause", "play", "playing", "progress", "ratechange",
+    "seeked", "seeking", "stalled", "suspend", "timeupdate", "volumechange",
+    "waiting", "toggle");
 
 /** Initialize all elements with py-* handlers attributes  */
 export function initHandlers(interpreter: InterpreterClient) {
     logger.debug('Initializing py-* event handlers...');
-    for (const pyAttribute of pyAttributeToEvent.keys()) {
-        createElementsWithEventListeners(interpreter, pyAttribute);
+    for (const browserEvent of browserEvents) {
+        createElementsWithEventListeners(interpreter, browserEvent);
     }
 }
 
 /** Initializes an element with the given py-on* attribute and its handler */
-function createElementsWithEventListeners(interpreter: InterpreterClient, pyAttribute: string) {
-    const matches: NodeListOf<HTMLElement> = document.querySelectorAll(`[${pyAttribute}]`);
+function createElementsWithEventListeners(interpreter: InterpreterClient, browserEvent: string) {
+    const matches: NodeListOf<HTMLElement> = document.querySelectorAll(`[py-${browserEvent}], [py-${browserEvent}-code]`);
+
+    const pyEval = interpreter.globals.get('eval')
+    const pyCallable = interpreter.globals.get('callable')
+    const pyDictClass = interpreter.globals.get('dict')
+
+    const localsDict = pyDictClass()
+
     for (const el of matches) {
-        // If the element doesn't have an id, let's add one automatically!
+        console.log('🌈 ?? el:', el)
+        // If the element doesn't have an id, let's add one automatically
         if (el.id.length === 0) {
             ensureUniqueId(el);
         }
-        const handlerCode = el.getAttribute(pyAttribute);
-        const event = pyAttributeToEvent.get(pyAttribute);
-
-        if (pyAttribute === 'pys-onClick' || pyAttribute === 'pys-onKeyDown') {
-            const msg =
-                `The attribute 'pys-onClick' and 'pys-onKeyDown' are deprecated. Please 'py-click="myFunction()"' ` +
-                ` or 'py-keydown="myFunction()"' instead.`;
-            createDeprecationWarning(msg, msg);
-            const source = `
-            from pyodide.ffi import create_proxy
-            Element("${el.id}").element.addEventListener("${event}",  create_proxy(${handlerCode}))
-            `;
-
-            // We meed to run the source code in a try/catch block, because
-            // the source code may contain a syntax error, which will cause
-            // the splashscreen to not be removed.
-            try {
-                interpreter.run(source);
-            } catch (e) {
-                logger.error((e as Error).message);
-            }
-        } else {
-            el.addEventListener(event, () => {
-                void (async () => {
+
+        if (el.getAttributeNames().find(s => s.includes('code'))) {
```

Comment:
```js
// this fails
['some-code', 'nope'].includes('code')

// this works
['code', 'nope'].includes('code')
```

but here you grab all attribute names to then check if the string includes code:

```js
// this works
'some-code'.includes('code')
```

So, once you know that some attribute has `code` in it, it could be any `whatever-code` but you don't check that, you just assign the event listener you create yourself, assuming that previous `if` contained that ... this is the confusion I have around that `if`, it checks for something not used after so I believe you want to create the string you are after, you want to verify if the attribute is present and, if that's the case, you want to add the listener ... do you think this makes sense?

Reply:
I think there is perhaps an even simpler option. Currently you do:
https://github.com/marimeireles/pyscript/blob/9c362677b669a3372c7c7e3b73a93e25aa09f09f/pyscriptjs/src/components/pyscript.ts#L95-103

And then within that loop determine whether you've got a match for `py-[browserEvent]` or `py-[browserEvent]-code`.

Why not just do:

```ts
matches = document.querySelectorAll(`[py-${browserEvent}]-code`);
for (const el of matches) {
    // py-event-code stuff
}

let matches: NodeListOf<HTMLElement> = document.querySelectorAll(`[py-${browserEvent}]`);
for (const el of matches) {
    // py-event stuff
}
```

Reply:
to be honest, if we were using [qsa-observer](https://github.com/WebReflection/qsa-observer#qsa-observer) all of this would be available for free just with an array of selectors and it will survive DOM mutations (bootstraps at distance, not only on pyscript loaded).